apps/judicial-system/backend/src/app/modules/repository/services/courtDocumentRepository.service.ts

```diff
@@ -158,6 +160,159 @@ export class CourtDocumentRepositoryService {
     }
   }
 
+  private async updateFiledDocumentsOrder({
+    documentOrder,
+    caseId,
+    courtSessionId,
+    courtDocumentId,
+    transaction,
+  }: {
+    documentOrder: number
+    caseId: string
+    courtSessionId: string
+    courtDocumentId: string
+    transaction?: Transaction
+  }): Promise<void> {
+    // Get the filed documents for the court session ordered by document order
+    const filedDocuments = await this.courtDocumentModel.findAll({
+      where: { caseId, courtSessionId },
+      order: [['documentOrder', 'ASC']],
+      transaction,
+    })
+
+    const currentDocument = filedDocuments.find((d) => d.id === courtDocumentId)
+
+    if (!currentDocument) {
+      throw new InternalServerErrorException(
+        `Could not find court document ${courtDocumentId} for court session ${courtSessionId} of case ${caseId}`,
+      )
+    }
+
+    const newOrder = documentOrder
+    const firstOrder = filedDocuments[0].documentOrder
+    const lastOrder = filedDocuments[filedDocuments.length - 1].documentOrder
+
+    // Validate the document order bounds
+    if (newOrder < firstOrder || newOrder > lastOrder) {
+      throw new BadRequestException(
+        `Order must be between ${firstOrder} and ${lastOrder}`,
+      )
+    }
+
+    const currentOrder = currentDocument.documentOrder
+
+    // Only adjust other documents if the document order is actually changing
+    if (currentOrder !== newOrder) {
+      if (newOrder > currentOrder) {
+        // Moving down: decrease order of documents between current and new position
+        await this.courtDocumentModel.update(
+          { documentOrder: literal('document_order - 1') },
+          {
+            where: {
+              caseId,
+              courtSessionId,
+              documentOrder: { [Op.gt]: currentOrder, [Op.lte]: newOrder },
+            },
+            transaction,
+          },
+        )
+      } else {
+        // Moving up: increase order of documents between new and current position
+        await this.courtDocumentModel.update(
+          { documentOrder: literal('document_order + 1') },
+          {
+            where: {
+              caseId,
+              courtSessionId,
+              documentOrder: { [Op.gte]: newOrder, [Op.lt]: currentOrder },
+            },
+            transaction,
+          },
+        )
+      }
+    }
+  }
+
+  private async updateMergedFiledDocumentsOrder({
```

Comment:
This function is almost identical to the `updateFiledDocumentsOrder` above, right? Can we remove one and have only one function?

Reply:
Cool, not a big deal for me, fine as is